file-formats/scp1/type/zif_aff_scp1_v1.intf.abap

```diff
@@ -0,0 +1,300 @@
+      "! Such data records are flagged as incomplete
+      incomplete TYPE ty_incomplete VALUE 'U',
+    END OF co_incomplete.
+
+  "! <p class="shorttext">Data Record Operation at Activation</p>
+  TYPES ty_operation_at_activation TYPE c LENGTH 1.
+
+  CONSTANTS:
+    "! <p class="shorttext">Data Record Operation at Activation</p>
+    BEGIN OF co_operation_at_activation,
+      "! <p class="shorttext">Insert/Update</p>
+      "! Data records which are to be inserted/updated at activation are flagged with value ''
+      insert_or_update TYPE ty_operation_at_activation VALUE '',
+      "! <p class="shorttext">Delete</p>
+      "! Data records which are to be deleted at activation are flagged with value 'L'
+      delete           TYPE ty_operation_at_activation VALUE 'L',
+    END OF co_operation_at_activation.
+
+  "! <p class="shorttext">Attribute of a field in a data record</p>
+  TYPES ty_additional_info TYPE c LENGTH 3.
+
+  CONSTANTS:
+    "! <p class="shorttext">Attribute of a field in a data record</p>
+    BEGIN OF co_field_attribute,
+      "! <p class="shorttext">Key field that is relevant and has Fixed value</p>
+      "! Fixed value for a key field. The value can not be changed after activation
+      fixed_key_field TYPE ty_additional_info VALUE 'FKY',
+      "! <p class="shorttext">Key field must be specified at activation</p>
+      "! Key field in which the user can change the value of the field when activating the data
+      key_field       TYPE ty_additional_info VALUE 'KEY',
+      "! <p class="shorttext">Value taken from BC Set at activation</p>
+      "! Default value of a data field
+      default_value   TYPE ty_additional_info VALUE 'USE',
+      "! <p class="shorttext">Value from BC Set is not modifiable in source table</p>
+      "! The value of this data field can no longer be changed after an activation
+      fixed_value     TYPE ty_additional_info VALUE 'FIX',
+      "! <p class="shorttext">Value must be specified at activation</p>
+      "! Data field whose value the user can change when activating
+      variable        TYPE ty_additional_info VALUE 'VAR',
+    END OF co_field_attribute.
+
+  TYPES:
+    "! <p class="shorttext">Data</p>
+    "! Data record represented in Field-Value pair
+    BEGIN OF ty_data,
+      "! <p class="shorttext">Field Name</p>
+      "! Name of data record field
+      "! $required
+      field_name  TYPE c LENGTH 30,
+      "! <p class="shorttext">Value</p>
+      "! Value of data record field
+      field_value TYPE string,
+    END OF ty_data.
+
+  "! <p class="shorttext">Data</p>
+  TYPES ty_row_data TYPE SORTED TABLE OF ty_data WITH UNIQUE KEY field_name.
+
+  TYPES:
+    "! <p class="shorttext">Selected Language</p>
+    "! Details of selected language
+    BEGIN OF ty_translation,
+      "! <p class="shorttext">Language</p>
+      "! Language in which the data record is translated
+      language    TYPE c LENGTH 1,
```

Comment:
I suggest to use `sy-langu` as type. This will convert language fields to ISO standard (similar to original language)

Reply:
As suggested, sy-langu has been used.